### ISSUE TITLE
Why is the postfix startup script sleeping for four years?

### ISSUE THREAD

**Issue by fbartels** [mailcow/mailcow-dockerized#4141]
**Prior to placing the issue, please check following:** *(fill out each checkbox with an `X` once done)*
- [x] I understand that not following or deleting the below instructions will result in immediate closure and/or deletion of my issue.
- [x] I have understood that this bug report is dedicated for bugs, and not for support-related inquiries.
- [x] I have understood that answers are voluntary and community-driven, and not commercial support.
- [x] I have verified that my issue has not been already answered in the past. I also checked previous [issues](https://github.com/mailcow/mailcow-dockerized/issues).

## Summary

The entrypoint of the postfix container has a really long sleep command at the end. I would think its better to run postfix in the foreground instead of keeping the container running by sleeping for four years.

https://github.com/mailcow/mailcow-dockerized/blob/d90d4f9640ed050386df8b88167a3a87c735e369/data/Dockerfiles/postfix/postfix.sh#L403

## Logs

I don't have logs.

## Reproduction

Run the container see sleep running, wait for years and the container will stop.

## System information

Not running the container, simply reviewing its code.

| Question | Answer |
| --- | --- |
| My operating system | Not relevant to the issue |
| Is Apparmor, SELinux or similar active? | Not relevant to the issue |
| Virtualization technlogy (KVM, VMware, Xen, etc - **LXC and OpenVZ are not supported** | Not relevant to the issue |
| Server/VM specifications (Memory, CPU Cores) | Not relevant to the issue |
| Docker Version (`docker version`) | Not relevant to the issue |
| Docker-Compose Version (`docker-compose version`) | Not relevant to the issue |
| Reverse proxy (custom solution) | Not relevant to the issue |

- Output of `git diff origin/master`, any other changes to the code? If so, **please post them**.
  - no changes to the code 
- All third-party firewalls and custom iptables rules are unsupported. *Please check the Docker docs about how to use Docker with your own ruleset*. Nevertheless, iptabels output can help us to help you: `iptables -L -vn`, `ip6tables -L -vn`, `iptables -L -vn -t nat` and `ip6tables -L -vn -t nat`.
- DNS problems? Please run `docker exec -it $(docker ps -qf name=acme-mailcow) dig +short stackoverflow.com @172.22.1.254` (set the IP accordingly, if you changed the internal mailcow network) and post the output.

---

Original message:

Hi Mailcow team,

I was looking through your Dockerfile for Postfix and noticed something that seemed odd to me.

In https://github.com/mailcow/mailcow-dockerized/blob/d90d4f9640ed050386df8b88167a3a87c735e369/data/Dockerfiles/postfix/postfix.sh#L403 the entrypoint script sleeps for 126144000 seconds (for years). While this is a long time and with applying updates to the host is unlikely to be reached I am wondering why it was implemented like this.

A better approach seems to be to start postfix in the foreground instead. e.g. `/usr/sbin/postfix -c /etc/postfix start-fg`

**Comment by andryyy** [comment]
Please read the issue template and don't post these questions here. :/

Check what else starts in the container. We need more than one process.

**Comment by fbartels** [comment]
@andryyy sorry, for whatever reason I did not get the template selection yesterday, when initially opening the issue page. Today it indeed suggests the templates.

I did not try to run your container, but looking through the entrypoint postfix is the only process running in the end (apart from sleep). If you want to keep the sleep command that is of course fine, but why not let it sleep for infinity?

```
sleep infinity
```
